pkg/apiserver/utils/binary_plan.go

```diff
@@ -50,10 +146,649 @@ func GenerateBinaryPlanJSON(b string) (string, error) {
+	}
+
+	diagOp, err := diagnosticOperatorNodes(node.Get(Children), diagOp)
+
+	// marked rows estimation error too high
+	if diagOp.needUdateStatistics {
+		switch operator {
+		case IndexFullScan, IndexRangeScan, TableFullScan, TableRangeScan, TableRowIDScan, Selection:
+			actRows := node.Get(ActRows).MustFloat64()
+			estRows := node.Get(EstRows).MustFloat64()
+			if actRows == 0 || estRows == 0 {
+				actRows = +1
+				estRows = +1
+			}
+			if actRows/estRows > 100 || estRows/actRows > 100 {
+				diagnosis = append(diagnosis, "The estimation error is high. Consider checking the health state of the statistics.")
+			}
+			diagOp.needUdateStatistics = true
+		default:
+			diagOp.needUdateStatistics = false
+		}
+	}
+
+	switch operator {
+	// index join
+	case IndexJoin, IndexMergeJoin, IndexHashJoin:
+		// only use in build
+		rootGroupInfo := node.Get(RootGroupExecInfo)
+		rootGroupInfoCount := len(rootGroupInfo.MustArray())
+		if rootGroupInfoCount > 0 {
+			for i := 0; i < rootGroupInfoCount; i++ {
+				joinTaskCountStr := rootGroupInfo.GetIndex(i).GetPath("inner", "task").MustString()
+				joinTaskCount, _ := strconv.Atoi(joinTaskCountStr)
+				if joinTaskCount > JoinTaskThreshold {
+					diagnosis = append(diagnosis, "This index join has a large build side. It might be slow and cause heavy pressure on TiKV. Consider using the optimizer hints to guide the optimizer to choose hash join.")
+					break
+				}
+			}
+		}
+	// unreasonable index return table plan
+	case IndexLookUpReader:
+		cNode := getBuildChildrenWithDriverSide(node, "build")
+		if getOperatorType(cNode) != Selection {
+			break
+		}
+		if len(cNode.Get(Children).MustArray()) != 1 {
+			break
+		}
+		gNode := cNode.Get(Children).GetIndex(0)
+
+		if getOperatorType(gNode) != IndexFullScan {
+			break
+		}
+
+		if !strings.Contains(gNode.Get(OperatorInfo).MustString(), "keep order:false") {
+			break
+		}
+
+		cNodeActRows := cNode.Get(ActRows).MustFloat64()
+		gNodeActRows := gNode.Get(ActRows).MustFloat64()
+		if cNodeActRows/gNodeActRows > returnTableThreshold {
+			diagnosis = append(diagnosis, "This IndexLookup read a lot of data from the index side. It might be slow and cause heavy pressure on TiKV. Consider using the optimizer hints to guide the optimizer to choose a better index or not to use index.")
+		}
+	case Selection:
+		if len(node.Get(Children).MustArray()) != 1 {
+			break
+		}
+		cNode := node.Get(Children).GetIndex(0)
+
+		if getOperatorType(cNode) != IndexFullScan {
+			break
+		}
+		if node.Get(StoreType).MustString() != "tikv" {
+			break
+		}
+
+		useOperator := false // operatror :  eq/ge/gt/le/lt/isnull/in
+		for _, op := range needCheckOperator {
+			if strings.Contains(operatorInfo, op) {
+				n := strings.Count(operatorInfo, op+"(")
+
+				useOperator = true
+				operatorInfo = strings.Replace(operatorInfo, op+"(", "", n)
+				operatorInfo = strings.Replace(operatorInfo, ")", "", n)
+			}
+		}
```

Comment:
Would you please describe the operator info's state transition to help the developer understand them in the future?